build.gradle

```diff
@@ -54,6 +57,25 @@ allprojects {
     configFile = new File(rootDir, "checkstyle/checkstyle.xml")
   }
 
+  task createConfigDocDir() {
+    outputs.dir configDocDir
+    file(configDocDir).mkdirs()
+  }
+
+  task createConfigDocs( dependsOn : ['createConfigDocDir', 'compileJava'], type : JavaExec) {
```

Comment:
It seems that we don't need createConfigDocDir since the ConfigDocumentationGenerator would create the directory if it doesn't exist. Also, does this task need to depend on `compileJava`? I tested the patch without this dependency and it still works.